pkg/arn/arn.go

```diff
@@ -62,6 +62,8 @@ func checkPartition(partition string) error {
 	case endpoints.AwsPartitionID:
 	case endpoints.AwsCnPartitionID:
 	case endpoints.AwsUsGovPartitionID:
+	case endpoints.AwsIsoPartitionID:
```

Comment:
Can you modify this function so instead of enumerating partitions, we just check all the ones the SDK knows about? Check out [`endpoints.DefaultPartitions()`](https://docs.aws.amazon.com/sdk-for-go/api/aws/endpoints/#DefaultPartitions)

Reply:
+1, Feedback addressed.

```
=== RUN   TestUserARN
--- PASS: TestUserARN (0.00s)
PASS
```